dsl_parser/rel_graph.py

```diff
@@ -741,8 +742,8 @@ def _node_instance_id(node_id, ctx):
     return new_node_instance_id
 
 
-def _generate_id():
-    return '%05x' % random.randrange(16 ** 5)
+def _generate_id(id_len=5):
+    return ''.join(choice(digits + ascii_letters) for _ in xrange(id_len))
```

Comment:
the code above uses two different approaches to concatenate strings using str.join and '+'
in general, I'd recommend to avoid the '+' for string concatenation as it's inefficient.
how about: `''.join(choice(''.join([digits, ascii_letters])) for _ in xrange(id_len))` or `''.join(sample(''.join([ascii_letters, digits]), id_len))` if you import sample instead of choice